tests/pytests/unit/grains/test_core.py

```diff
@@ -3623,3 +3871,1154 @@ def _mock_is_file(filename):
+    """
+    mock_gpudata = """
+Graphics/Displays:
+
+    NVIDIA GeForce 320M:
+
+      Chipset Model: NVIDIA GeForce 320M
+      Type: GPU
+      VRAM (Total): 256 MB
+      Vendor: NVIDIA (0x10de)
+      Device ID: 0x08a0
+      Revision ID: 0x00a2
+      ROM Revision: 3533
+      Displays:
+        Color LCD:
+          Display Type: LCD
+          Resolution: 1280 x 800
+          UI Looks like: 1280 x 800
+          Framebuffer Depth: 24-Bit Color (ARGB8888)
+          Main Display: Yes
+          Mirror: Off
+          Online: Yes
+          Automatically Adjust Brightness: Yes
+          Connection Type: Internal
+
+"""
+    with patch.dict(core.__salt__, {"cmd.run": MagicMock(return_value=mock_gpudata)}):
+        ret = core._osx_gpudata()
+        assert ret["num_gpus"] == 1
+        assert ret["gpus"] == [{"vendor": "nvidia", "model": "GeForce 320M"}]
+
+    with patch.dict(core.__salt__, {"cmd.run": MagicMock(side_effect=OSError)}):
+        ret = core._osx_gpudata()
+        assert ret == {"num_gpus": 0, "gpus": []}
+
+
+def test_get_master():
+    """
+    test get_master
+    """
+    ret = core.get_master()
+    assert "master" in ret
+
+    with patch("salt.grains.core.__opts__", {"master": "test_master_id"}):
+        ret = core.get_master()
+        assert "master" in ret
+        assert ret["master"] == "test_master_id"
+
+
+def test__selinux():
+    """
+    test _selinux
+    """
+    with patch.dict(
+        core.__salt__,
+        {
+            "cmd.run": MagicMock(return_value="Enforcing"),
+            "cmd.retcode": MagicMock(return_value=1),
+        },
+    ), patch.object(core, "_linux_bin_exists", MagicMock(return_value=False)):
+        ret = core._selinux()
+        assert ret == {"enabled": False}
+
+    with patch.dict(
+        core.__salt__,
+        {
+            "cmd.run": MagicMock(return_value="Enforcing"),
+            "cmd.retcode": MagicMock(return_value=0),
+        },
+    ), patch.object(core, "_linux_bin_exists", MagicMock(return_value=True)):
+        ret = core._selinux()
+        assert ret == {"enabled": True, "enforced": "Enforcing"}
+
+    with patch.dict(
+        core.__salt__,
+        {
+            "cmd.run": MagicMock(return_value="Disabled"),
+            "cmd.retcode": MagicMock(return_value=0),
+        },
+    ), patch.object(core, "_linux_bin_exists", MagicMock(return_value=True)):
+        ret = core._selinux()
+        assert ret == {"enabled": True, "enforced": "Disabled"}
+
+
+def test__systemd():
+    """
+    test _systemd
+    """
+    with patch.dict(
+        core.__salt__,
+        {
+            "cmd.run": MagicMock(
+                return_value="systemd 254 (254.3-1)\n+PAM +AUDIT -SELINUX -APPARMOR -IMA +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBFDISK +PCRE2 -PWQUALITY +P11KIT -QRENCODE +TPM2 +BZIP2 +LZ4 +XZ +ZLIB +ZSTD +BPF_FRAMEWORK +XKBCOMMON +UTMP -SYSVINIT default-hierarchy=unified"
```

Comment:
```suggestion
                return_value=(
                    "systemd 254 (254.3-1)\n+PAM +AUDIT -SELINUX -APPARMOR -IMA +SMACK "
                    "+SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS "
                    "+FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBFDISK +PCRE2 "
                    "-PWQUALITY +P11KIT -QRENCODE +TPM2 +BZIP2 +LZ4 +XZ +ZLIB +ZSTD "
                    "+BPF_FRAMEWORK +XKBCOMMON +UTMP -SYSVINIT default-hierarchy=unified"
                )
```
This will make it more readable when on a file editor...